### ISSUE TITLE
low compression ratio

### ISSUE THREAD

**Issue by jpillora** [ulikunitz/xz#5]
```
$ ll
-rw-r--r--  1 jpillora  wheel   550K 30 Jan 21:18 a.log
$ cp a.log b.log
$ cp a.log c.log
$ xz a.log
$ gxz b.log
$ gzip c.log
$ ll
-rw-r--r--  1 jpillora  wheel   6.0K 30 Jan 15:43 a.log.xz
-rw-r--r--  1 jpillora  wheel   207K 30 Jan 21:16 b.log.xz
-rw-r--r--  1 jpillora  wheel    10K 30 Jan 21:16 c.log.gz
```

Any idea why this is?

**Comment by kofalt** [comment]
Would it be possible to provide the file in question?

**Comment by jpillora** [comment]
Yep, will post once I get home

**Comment by jpillora** [comment]
redacted copy with similar results: https://gist.github.com/jpillora/f8e0d0b3eb6a066ae5b0/raw/84847dae70ba396d7a2d19ba888e756a7b37362f/a.log

**Comment by ulikunitz** [comment]
Hi, many thanks for reporting the issue.

I have stated in the README.md that the compression ratio is not as good as with xz, so this is not a surprise. The way of selecting matches has been poor leading to a bad compression ratio.  I have worked on it over the last few days. Now the dev branch is compressing your file to 6.8 KiB, which is still not as good than the xz tool but better than gzip. Since compression is now much faster than the xz tool I'm optimistic that I can trade compression speed  to improve the compression ratio. But this requires still some work and tooling.

I have marked the issue for closure by the v0.5 milestone.
